src/core/public/application/application_service.test.ts

```diff
@@ -580,14 +580,23 @@ describe('#start()', () => {
 
     it('creates URLs with path parameter', async () => {
       service.setup(setupDeps);
-
       const { getUrlForApp } = await service.start(startDeps);
 
       expect(getUrlForApp('app1', { path: 'deep/link' })).toBe('/app/app1/deep/link');
       expect(getUrlForApp('app1', { path: '/deep//link/' })).toBe('/app/app1/deep/link');
       expect(getUrlForApp('app1', { path: '//deep/link//' })).toBe('/app/app1/deep/link');
       expect(getUrlForApp('app1', { path: 'deep/link///' })).toBe('/app/app1/deep/link');
     });
+
+    it('creates absolute URLs when `absolute` parameter is true', async () => {
+      service.setup(setupDeps);
+      const { getUrlForApp } = await service.start(startDeps);
+
+      expect(getUrlForApp('app1', { absolute: true })).toBe('http://localhost/app/app1');
+      expect(getUrlForApp('app2', { path: 'deep/link', absolute: true })).toBe(
+        'http://localhost/app/app2/deep/link'
+      );
+    });
```

Comment:
Because of current implementation, we cannot properly mock the 'location provider' to test this, so the actual protocol and domains are the jsdom's default: `http:` and `localhost`. Should I add a functional test to test the behavior in a real browser environ?

Reply:
it seems that `jsdom` doesn't provide an easy way to mock it. I'd rather skip it then, since `The behavior is specified, this is not relying on internal browser implementation`